### ISSUE TITLE
Windows Integrated shell includes double quote in sent arguments

### ISSUE THREAD

**Issue by george-haddad** [Microsoft/vscode#34290]
- VSCode Version: Code 1.16.0 (787b31c0474e6165390b5a5989c9619e3e16f953, 2017-09-06T16:27:49.891Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
project-manager|ale|0.21.1
npm-intellisense|chr|1.3.0
path-intellisense|chr|1.4.2
vscode-markdownlint|Dav|0.10.1
vscode-eslint|dba|1.3.1
vscode-instant-markdown|dba|1.3.0
githistory|don|0.2.3
gitlens|eam|5.0.0
vscode-npm-script|eg2|0.2.1
prettier-vscode|esb|0.23.1
flow-for-vscode|flo|0.7.1
vscode-docker|Pet|0.0.16
vscode-icons|rob|7.12.0
JavaScriptSnippets|xab|1.4.1
ReactSnippets|xab|1.4.0

---

Steps to Reproduce:

1. Open user settings
2. Add the following settings to initialize the integrated command prompt with nodejs env
```javascript
  "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\CMD.exe",
  "terminal.integrated.shellArgs.windows": [
    "/K C:\\Progra~1\\nodejs\\nodevars.bat && set NODE_ENV=development"
  ]
```
3. Save the file
4. Press `CTRL+~` to get the command prompt up
5. Execute command `echo %NODE_ENV%`
6. It will display the string including the trailing double quote
```
Your environment has been set up for using Node.js 8.2.1 (x64) and npm.

D:\git\myproject>echo %NODE_ENV%
development"

D:\git\myproject>
```
7. It should not include the double quote
8. Workaround is to include a dummy env var that no one will ever use
```
"/K C:\\Progra~1\\nodejs\\nodevars.bat && set NODE_ENV=development && set QQQ="
```

Reproduces without extensions: Yes

**Comment by Neutrino-Sunset** [comment]
Thanks for the workaround.

**Comment by bowerscd** [comment]
there seems to be a bit more here than just the dangling quote:

for example, try with:

```json
    "terminal.integrated.shellArgs.windows": [
        "/K",
        "alias -v BCZS \"build -cZP $*\" & python -c \"print('\\a')\" "
    ]
```

what should be executed is:

`alias -v BCZS "build -cZP $*" & python -c "print('\a')"`

which should be parsed as: 

`[/K, alias, -v, BCZS, build -cZP $*, &, python, -c, print('\a')]`

however the tokens sent to /K are:

`[alias, -v, BCZS, build, -cZP, $*, &, python, -c, print('\a')]`


A work around would be to wrap this up into a native shell script and just /K the shell script.

**Comment by Tyriar** [comment]
If anyone wants to look into this, here is the function that runs on Windows to convert the string to a Windows "Command Line" string which is used to launch the process.

**Comment by george-haddad** [comment]
@Tyriar I think you're missing a link in your comment :)

**Comment by Tyriar** [comment]
@george-haddad someone's paying attention 😉 https://github.com/Microsoft/node-pty/blob/9faf0fff3958b8c8e79e60a18fe31013f9bc05e7/src/windowsPtyAgent.ts#L116

**Comment by bowerscd** [comment]
I tested this with some command line tests. The result of 

```ts 
console.log(argsToCommandLine("cmd.exe", "/K alias -v BCZS \"build -cZP $*\" & python -c \"print('\\a')\""));
```

results in 

```
cmd.exe /K alias -v BCZS "build -cZP $*" & python -c "print('\a')"
```

which is correct & what we expect for the issue I reported - it must be either how this function receives the arguments or how we pass the arguments to the shell. 

---

Addressing George's issue, attempting the same function with the args:

```ts
argsToCommandLine("cmd.exe", "/K C:\\Progra~1\\nodejs\\nodevars.bat && set NODE_ENV=development")
```
results in: `cmd.exe /K C:\Progra~1\nodejs\nodevars.bat && set NODE_ENV=development`

I copied and pasted that into run - we get this (as expected) 

```CMD
Your environment has been set up for using Node.js 8.9.4 (x64) and npm.

C:\Windows\System32>echo %NODE_ENV%
development

```

---

TL;DR the linked function seems to be working just fine, when it comes to making these args into a command line.

**Comment by oopsaleem** [comment]
Try
```
    "terminal.integrated.shell.windows": "C:\\Windows\\System32\\cmd.exe",
    "terminal.integrated.shellArgs.windows": [
        "/k",
        "C:\\Program Files\\nodejs\\nodevars.bat"
    ],
```

**Comment by alexr00** [comment]
If you do:
`    "terminal.integrated.shellArgs.windows": 
    [
        "/K",
        "C:\\Progra~1\\nodejs\\nodevars.bat && set NODE_ENV=development"
    ]`
or:
`    "terminal.integrated.shellArgs.windows": 
        "/K C:\\Progra~1\\nodejs\\nodevars.bat && set NODE_ENV=development",`
That should work around the trailing quote.
